codex-rs/core/src/codex.rs

```diff
@@ -829,6 +829,79 @@ async fn submission_loop(
                     }
                 });
             }
+            Op::SummarizeContext => {
+                let sess = match sess.as_ref() {
+                    Some(sess) => sess,
+                    None => {
+                        send_no_session_event(sub.id).await;
+                        continue;
+                    }
+                };
+
+                // Create a summarization request as user input
+                const SUMMARIZATION_PROMPT: &str = r#"
```

Comment:
Let's put this in an `.md` file and `include_str!()` so it's easier to maintain.